pages/glossary/assigned-cell.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Assigned cell
+key: assigned-cell
+unambiguous: true
+objective: true
+input_aspects:
+  - Accessibility tree
+  - DOM tree
+---
+
+This definition of assigning header cells to data cells is different from the [internal algorithm for scanning and assigning header cells][https://html.spec.whatwg.org/multipage/tables.html#internal-algorithm-for-scanning-and-assigning-header-cells] in the [HTML specification][https://html.spec.whatwg.org/]. The algorithm provides guidance on how those elements should be mapped to accessibility APIs whereas this definition focuses solely on the outcome of the mapping. It takes into consideration the structure of the [table][] and [grid][] elements as they appear in the accessibility tree.
+
+In order for an [accessible object][] with the [semantic role][] of [rowheader][] to be assigned to an [accessible object][] with the [semantic role][] of either [cell][], [gridcell][], or inheriting from [cell][], at least one of the following is true:
+
+- there is at least one non-empty [accessible object][] with a [semantic role][] of either [cell][], or inheriting from [cell][], that has the `headers` attribute with the value that matches the `id` [attribute value][] of the [rowheader][] that is being evaluated, and it is an [owned element][] of the same [table][] or [grid][] as the [rowheader][]; or
+- there is at least one non-empty [accessible object][] with a [semantic role][] of either [cell][], or inheriting from [cell][], that is a child an [accessible object][] with the [semantic role][] of [row][] that is also the parent of the evaluated [rowheader][];
+
+An [accessible object][] with the [semantic role][] of [columnheader][] is assigned to an element with a [semantic role][] of either [cell][], or inheriting from [cell][], if at least one of the following is true:
+
+- there is at least one non-empty [accessible object][] with a [semantic role][] of either [cell][], or inheriting from [cell][], that has the `headers` attribute with the value that matches the `id` [attribute value][] of the [columnheader][] that is being evaluated, and it is an [owned element][] of the same [table][] or [grid][] but it is not a child of the same [accessible object][] with the [semantic role][] of [row][] as the [columnheader][]; or
+- there is at least one non-empty [accessible object][] with a [semantic role][] of either [cell][], or inheriting from [cell][], that is a child of an [accessible object][] with the [semantic role][] of [row][], that is not the parent element of the [columnheader][] for which the assignment is evaluated; the [cell][]'s [row][] is an [owned element][] of the same [table][] or [grid][] to which the [columnheader][] belongs and has the same [child index][] as the [columnheader][], or it has a `colspan` attribute which together with the [cell][]'s [child index][] value spans over the [child index][] value of the [columnheader][];
```

Comment:
As for the first example with a table cell that spans over two rows and columns, using NVDA on both Chrome and FF associates the `<td>Bar</td>` to `<th>H3</th>`.  Current definition and the accessibility tree that I was looking at it did not foresee such outcome.

In the second example with `aria-hidden`, the `<td>D3</td>` is assigned to H3 in FF but not in Chrome (in Chrome it's H2). I'm not sure at the moment what might be causing it.

As for the third example, that is currently a failure as the cells are not owned by a `row`.  Interestingly, FF and NVDA do not associate those cells at all whereas Chrome does.

In summary, there are examples that are clearly a failure but the test target that is failing is different depending on the browser. I'm not sure if that is a deal breaker or not since implemenations should fail such example anyway. I also need to take into consideration how `rowspan` and `colspan` affects association when the number of cells in a row is not consistent. I'd need to test that with Jaws and VO as well to get a better understanding.